src/main/java/com/unstoppabledomains/resolution/Resolution.java

```diff
@@ -1,169 +1,217 @@
-     * @return etherium address of a domain's getOwner
-     */
-    @Deprecated
+    @Override
     public String owner(String domain) throws NamingServiceException {
         NamingService service = findService(domain);
         return service.getOwner(domain);
     }
-    
+
+    private Map<NamingServiceType, NamingService> getServices(String blockchainProviderUrl) {
+        return new HashMap<NamingServiceType, NamingService>() {{
+            put(NamingServiceType.CNS, new CNS(new NSConfig(Network.MAINNET, blockchainProviderUrl)));
+            put(NamingServiceType.ENS, new ENS(new NSConfig(Network.MAINNET, blockchainProviderUrl)));
+            put(NamingServiceType.ZNS, new ZNS(new NSConfig(Network.MAINNET, ZILLIQA_DEFAULT_URL)));
+        }};
+    }
+
     private NamingService findService(String domain) throws NamingServiceException {
-        for (NamingService service : services) {
+        for (NamingService service : services.values()) {
             if (Boolean.TRUE.equals(service.isSupported(domain))) return service;
         }
         throw new NamingServiceException(NSExceptionCode.UnsupportedDomain, new NSExceptionParams("d", domain));
     }
 
-    private NamingService[] buildServices(String providerUrl) {
-        NamingService[] services = new NamingService[3];
-        services[0] = new CNS(providerUrl);
-        services[1] = new ENS(providerUrl);
-        services[2] = new ZNS("https://api.zilliqa.com");
-        return services;
+    public static class Builder {
+        private static final String INFURA_URL = "https://%s.infura.io/v3/%s";
+
+        private final Map<NamingServiceType, NSConfig> serviceConfigs;
+
+        private Builder() {
+            serviceConfigs = new HashMap<NamingServiceType, NSConfig>() {{
+                put(NamingServiceType.CNS, new NSConfig(Network.MAINNET, LINKPOOL_DEFAULT_URL));
+                put(NamingServiceType.ENS, new NSConfig(Network.MAINNET, LINKPOOL_DEFAULT_URL));
+                put(NamingServiceType.ZNS, new NSConfig(Network.MAINNET, ZILLIQA_DEFAULT_URL));
+            }};
+        }
+
+        /**
+         * @param nsType  the naming service for which config is applied
+         * @param chainId blockchain network ID
+         * @return builder object to allow chaining
+         */
+        public Builder chainId(NamingServiceType nsType, Network chainId) {
+            NSConfig nsConfig = serviceConfigs.get(nsType);
+            nsConfig.setChainId(chainId);
+            return this;
+        }
+
+        /**
+         * @param nsType      the naming service for which config is applied
+         * @param providerUrl blockchain provider URL
+         * @return builder object to allow chaining
+         */
+        public Builder providerUrl(NamingServiceType nsType, String providerUrl) {
+            NSConfig nsConfig = serviceConfigs.get(nsType);
+            nsConfig.setBlockchainProviderUrl(providerUrl);
+            return this;
+        }
```

Comment:
Since this PR renames several methods to follow Java naming conventions, should we introduce `Builder` with `get`/`set` prefixes in method names as well?

Reply:
Sounds reasonable, but usually Builder is some kind of exception from these rules :) 
I was following lombok-style implementation: https://projectlombok.org/features/Builder